### PR TITLE
Close farmer properly

### LINKED ISSUE #26: Close farmer gracefully
Right now farmer task gets aborted, while we need to make sure that after calling `.close` that farmer exited gracefully and freed all of its resources.

### PR DESCRIPTION
**Pull request by i1i1**
Closes #26

Depends on https://github.com/subspace/subspace/pull/910

### CODE REVIEW

File: src/farmer.rs
Comment (ozgunozerk) [inline src/farmer.rs:10]: nice, switched to oneshot :)

File: src/farmer.rs
Comment (ozgunozerk) [inline src/farmer.rs:462]: seems like a good place to use the new `let else` to me :) 
Feel free to ignore though, not a critical suggestion
```suggestion
        let Some(cmd_sender) = maybe_cmd_sender.take() else {
            return;
        }
```

**Review by ozgunozerk — DISMISSED** [review]
lgtm

**Comment by ozgunozerk** [comment]
for the polishing ideas in the future, do you want to separate `plot` related things into another file @i1i1 ? So the `farmer.rs` will look more organized?

File: src/farmer.rs
Comment (i1i1) [reply src/farmer.rs:10]: Yeah, right :D

**Comment by i1i1** [comment]
> for the polishing ideas in the future, do you want to separate plot related things into another file @i1i1 ? So the farmer.rs will look more organized?

Yeah, why not. Currently the file is not that big, so we can keep it in the same place.

**Review by ozgunozerk — APPROVED** [review]

**Comment by i1i1** [comment]
Please don't merge it, until the merge of the monorepo pr. Nothing will change though apart from hash of commit.

**Comment by i1i1** [comment]
It will take some time for monorepo to merge the update substrate pr, so I guess it is fine to merge it as it is.

**Pull request merged**
